### PR TITLE
needs reviewed to merge resolves #34

### LINKED ISSUE #34: Refactor GameService to use data layer
Use the `Game` Eloquent model in all of the service methods rather than hard-coded data. for help with queries see: 
https://laravel.com/docs/9.x/eloquent#retrieving-models
https://laravel.com/docs/9.x/collections#available-methods
- getGames: should return games alphabetically by title. use the `sortBy` and `all` methods.
- getGameById: should return a single game, use the `find` method.
- searchGamesByTitle: should use a `where like` clause to filter results by title.

### PR DESCRIPTION
**Pull request by 12bitsample**
Okay so I am getting an error when I click games on the site, but I believe this is because the seeders haven't been merged yet maybe? Just want to confirm the logic in the methods inside of GameService looks sound before getting an approved PR and merging. Thanks!

### CODE REVIEW

File: app/Services/GameService.php [hunk 1]
```diff
@@ -8,7 +8,7 @@ class GameService {
 
     function getGamesById (int $id)
     {
-        $games = $this->getGames();
+        $games = Game::find($id);
```
Comment (oldskoolfan) [inline app/Services/GameService.php:11]: this is all you should need, you can remove the foreach loop logic below. In order to make sure a NotFoundHttpException is thrown, i would suggest using the `findOrFail()` method instead.

File: app/Services/GameService.php [hunk 2]
```diff
@@ -21,12 +21,11 @@ function getGamesById (int $id)
     // todo: add optional params for orderBy, direction, and limit
     function getGames()
     {
-        return
-            [
-                Game::make(['id' => 1, 'title' => 'Monopoly', 'rating' => 1]),
-                Game::make(['id' => 2, 'title' => 'Risk', 'rating' => 2]),
-                Game::make([ 'id' => 3, 'title' => 'DND', 'rating' => 5]),
-            ];
+        $allGames = Game::all('id', 'title', 'rating', 'age', );
```
Comment (oldskoolfan) [inline app/Services/GameService.php:24]: your error is from selecting a column that doesn't exist. you don't need specify columns here, just use `Game::all()`

File: app/Services/GameService.php (hunk 1 above)
Comment (12bitsample) [reply app/Services/GameService.php:11]: Resolved!

File: app/Services/GameService.php [hunk 3]
```diff
@@ -8,25 +8,19 @@ class GameService {
...
-                Game::make([ 'id' => 3, 'title' => 'DND', 'rating' => 5]),
-            ];
+        $allGames = Game::all();
+        $sortedAllGames = $allGames->sortBy('title');
+
+        return $sortedAllGames;
+
     }
 
     function searchGamesByTitle (string $searchTitle)
```
Comment (oldskoolfan) [inline app/Services/GameService.php:26]: you need to update this method as well, as i mentioned in the issue description

Comment (12bitsample) [reply app/Services/GameService.php:25]: I believe this was resolved as of 11/7

Comment (oldskoolfan) [reply app/Services/GameService.php:27]: we can resolve it once you push your changes, right now this method is unchanged in your branch.

Comment (12bitsample) [reply app/Services/GameService.php:33]: Okay I think most recent push has resolved this!

File: database/seeders/DatabaseSeeder.php [hunk 4]
```diff
@@ -13,6 +14,7 @@ class DatabaseSeeder extends Seeder
      */
     public function run()
     {
+        GameFactory::factory(10)->create();
```
Comment (oldskoolfan) [inline database/seeders/DatabaseSeeder.php:17]: i don't think we need this here

Comment (12bitsample) [reply database/seeders/DatabaseSeeder.php]: removed and resolved in most recent push!

**Review by oldskoolfan — APPROVED** [review]

**Pull request merged**